tracer/test/Datadog.Trace.TestHelpers.AutoInstrumentation/TestHelper.cs

```diff
@@ -146,75 +146,30 @@ public Process StartSample(MockTracerAgent agent, string arguments, string packa
 
         public async Task<bool> TakeMemoryDump(Process process)
         {
-            // We don't know if procdump is available, so download it fresh
-            if (!EnvironmentTools.IsWindows())
-            {
-                Output.WriteLine("Not running on windows, skipping memory dump");
-                return false;
-            }
-
             try
             {
-                const string url = @"https://download.sysinternals.com/files/Procdump.zip";
-                var client = new HttpClient();
-                var zipFilePath = Path.GetTempFileName();
-                Output.WriteLine($"Downloading Procdump to '{zipFilePath}'");
-                using (var response = await client.GetAsync(url, HttpCompletionOption.ResponseHeadersRead))
+                if (EnvironmentTools.IsLinux())
                 {
-                    using var bodyStream = await response.Content.ReadAsStreamAsync();
-                    using Stream streamToWriteTo = File.Open(zipFilePath, FileMode.Create);
-                    await bodyStream.CopyToAsync(streamToWriteTo);
+                    var dotnetRuntimeFolder = Path.GetDirectoryName(typeof(object).Assembly.Location);
+                    var createDumpExecutable = Path.Combine(dotnetRuntimeFolder!, "createdump");
```

Comment:
Can we use `createdump` for Windows too?

Reply:
That's a great point! That should be possible, but it would make the code a bit more complex, since I'm not sure how we would find the path of of `createdump` in the `net461` scenario. AFAIK that would require [checking the Windows Registry](https://stackoverflow.com/a/62875634) to find the .NET Core runtime folder, so push comes to shuve not sure all-in-all this change would make the code shorter or easier :/

Do you have any thoughts on this?

Reply:
Oh, yes you have a point.
In that case, I'm not sure it will for `.netcoreapp2.1` and `.netcoreapp3.1` either, I don't think `createdump.exe` is published with those runtimes.

As possible solution - we can upload `createdump.exe` to some public artifact, and download it, as we do it for `procdump`

Anyway, I think we should add check for tool existence, before trying to create dump.

Reply:
Yeps, both those solutions seem totally valid, but I'm not sure they would actually make the code simpler / shorter / easier.

Actually, now that I think about it, there's another reason to keep `procdump`: it supports capturing a dump at the moment a specific first-chance-exception has occurred. I'm actually [planning to leverage that a designed branch](https://github.com/DataDog/dd-trace-dotnet/commit/515413cce7e04a7426d955b0aeba3643ac266cb1) in order to diagnose the flakey test (which I believe is exhibiting a real bug in our code) where the Serilog exploration test is randomly failing with a `CannotUnloadAppDomainException`.

Reply:
I see.
We still need to solve the problem for `.netcoreapp2.1` and `.netcoreapp3.1` on Linux.

Reply:
Sorry, I missed your earlier comment:
>In that case, I believe, your approach will not work for .netcoreapp2.1 and .netcoreapp3.1, too, since createdump.exe is not published with those runtimes.

That's actually not true. At least on Linux, `createdump` is indeed published with the latest versions of those runtimes. I've validated this both by looking at our CI logs, but also by running `ls` locally:
```
omerr@comp/usr/share/dotnet/shared/Microsoft.NETCore.App/2.1.30$ ls createdump
createdump
````

Reply:
Oh, Microsoft 😄 
They do include is for Linux, but do not for Windows. 😅